rfcs/RFC-Upload-From-Dropbox.md

```diff
@@ -0,0 +1,417 @@
+link is provided, the new link will be used. If the submission is not in an error state, this endpoint will
+return an error.
+
+Starting an upload causes the database to be updated with a new dataset. The dataset will set the initial state of the upload
+to waiting once it has been accepted and is in the upload queue.
+
+**Request:**
+
+| Parameter             | Description                                                                    |
+| --------------------- | ------------------------------------------------------------------------------ |
+| Link                  | a shared link to the file                                                      |
+| submission_id         | identifies the submission                                                      |
+| _Optional_ dataset_id | identifies the dataset being uploaded. Used to check the status of the upload. |
+
+**Response:**
+
+| Key        | Description                                                               |
+| ---------- | ------------------------------------------------------------------------- |
+| dataset_id | identifies the dataset the file will be associated with after validation. |
+| status     | Provides the current status of the upload                                 |
+
+**Error Responses:**
+
+| Code | Description                                                                               |
+| ---- | ----------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission |
+| 400  | if the file type is invalid.                                                              |
+| 409  | if there is an existing submissions in progress with the same dataset_id.                 |
+| 413  | if the links refer to a file that exceeds the max size allowed.                           |
+
+#### GET submission/{submission_id}/upload
+
+Checks the status of an existing upload job.
+
+**Request:**
+
+| Parameter     | Description                          |
+| ------------- | ------------------------------------ |
+| submission_id | identifies the submission            |
+| dataset_id    | identifies the dataset to be queried |
+
+**Response:**
+
+| Key      | Description                                   |
+| -------- | --------------------------------------------- |
+| status   | Provides the current status of the upload.    |
+| message  | If an error occurred, the message shows here. |
+| progress | The current progress of the upload.           |
+
+**Error Responses:**
+
+| Code | Description                                                                                                             |
+| ---- | ----------------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress upload. |
```

Comment:
_upload in-progress upload_? Do you mean _in-progress upload_?

Reply:
fixed